src/pages/ShopEditPage.tsx

```diff
@@ -1,3 +1,272 @@
+
+import { useNavigate } from "react-router-dom";
+
+import { SeoulDistrict, SeoulDistricts, ShopCategory } from "../types";
+
+import {
+  getPublicURL,
+  postImage,
+  putImage,
+} from "@/apis/services/imageService";
+import { getShop, putShop } from "@/apis/services/shopService";
+import { Camera, Close } from "@/assets/icon";
+import Button from "@/components/Button";
+import Select from "@/components/Select";
+import TextField from "@/components/TextField";
+import { useUserStore } from "@/hooks/useUserStore";
+import { extractDigits, numberCommaFormatter } from "@/utils/number";
+
+const CATEGORY_OPTIONS = [
+  { label: "한식", value: "한식" },
+  { label: "중식", value: "중식" },
+  { label: "일식", value: "일식" },
+  { label: "양식", value: "양식" },
+  { label: "분식", value: "분식" },
+  { label: "카페", value: "카페" },
+  { label: "편의점", value: "편의점" },
+  { label: "기타", value: "기타" },
+];
+
+type FormType = {
+  name: string;
+  category: ShopCategory;
+  address1: SeoulDistrict;
+  address2: string;
+  originalHourlyPay: string;
+  description: string;
+};
+
+const FIELD_LABELS: Record<keyof FormType, string> = {
+  name: "가게 이름",
+  category: "분류",
+  address1: "주소",
+  address2: "상세 주소",
+  originalHourlyPay: "기본 시급",
+  description: "가게 설명",
+};
+
+export default function ShopRegisterPage() {
+  const navigate = useNavigate();
+  const { user } = useUserStore();
+  const shopId = user?.shopId;
+  const fileInputRef = useRef<HTMLInputElement | null>(null);
+  const [imagePreview, setImagePreview] = useState<string | null>(null);
+  const [imageFile, setImageFile] = useState<File | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [form, setForm] = useState<FormType>({
+    name: "",
+    category: "" as ShopCategory,
+    address1: "" as SeoulDistrict,
+    address2: "",
+    originalHourlyPay: "",
+    description: "",
+  });
+
+  useEffect(() => {
+    async function fetchShop() {
+      if (!shopId) return;
+      const res = await getShop(shopId);
+      const shop = res.data.item;
+      setForm({
+        name: shop.name,
+        category: shop.category,
+        address1: shop.address1,
+        address2: shop.address2,
+        originalHourlyPay: numberCommaFormatter(shop.originalHourlyPay),
+        description: shop.description,
+      });
+      if (shop.imageUrl) setImagePreview(shop.imageUrl);
+    }
+    fetchShop();
+  }, [shopId]);
+
+  const handleChange = (
+    key: keyof FormType,
+    value: string | SeoulDistrict | ShopCategory,
+  ) => {
+    setForm((prev) => ({ ...prev, [key]: value }));
+  };
+
+  const handleImageUpload = (e: ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+    if (!file) return;
+
+    setImageFile(file);
+
+    const reader = new FileReader();
+    reader.onload = () => {
+      if (typeof reader.result === "string") {
+        setImagePreview(reader.result);
+      }
+    };
+    reader.readAsDataURL(file);
```

Comment:
`URL.createObjectURL` API를 활용한 코드가
위->아래 흐름으로 더 보기 쉬운 것 같아요! 🤔
```suggestion
    // 이미 메모리에 blobURL이 존재한다면 해제하기
    if(imagePreview) {
      URL.revokeObjectURL(imagePreview);
    }
    // blobURL 생성하기
    const blobUrl = URL.createObjectURL(file);
    setImagePreview(blobUrl);